lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStartHandler.java

```diff
@@ -73,10 +106,20 @@ public void onNext(final StartTime startTime) {
   }
 
   private void onRestart(final StartTime startTime) {
-    if (restartHandlers.isPresent()) {
-      for (EventHandler<StartTime> restartHandler : this.restartHandlers.get()) {
+    if (this.driverRestartManager.isPresent() && this.restartHandlers.isPresent()) {
+      for (EventHandler<StartTime> serviceRestartHandler : this.serviceRestartHandlers.get()) {
+        serviceRestartHandler.onNext(startTime);
+      }
+
+      for (EventHandler<StartTime> restartHandler : this.restartHandlers.get()){
         restartHandler.onNext(startTime);
       }
+
+      // This can only be called after calling client restart handlers because REEF.NET
+      // JobDriver requires making this call to set up the InterOp handlers.
+      final RestartEvaluatorInfo restartEvaluatorInfo = this.driverRestartManager.get().onRestartRecoverEvaluators();
```

Comment:
The reason I did this was to make the separation more clear, such that a new implementation knows exactly what to do. If we merge all the functions into `onDriverRestart()`, I believe it'll be less clear as to what the implementer needs to do. Sure, we can rename the functions as well as the comments to make it explicit, but I think splitting the methods apart is much more obvious. @markusweimer what do you think?